remoteworkitem/flatten.go

```diff
@@ -0,0 +1,56 @@
+package remoteworkitem
+
+import (
+	"reflect"
+	"strings"
+)
+
+type stack []string
+
+func (s *stack) push(v string) {
+	*s = append(*s, v)
+}
+
+func (s *stack) pop() string {
+	l := len(*s)
+	if l == 0 {
+		return ""
+	}
+	stackReference := *s // because *s[:l-1] did not work.
+	poppedData := stackReference[l-1]
+	*s = stackReference[:l-1]
+	return poppedData
+}
+
+func (s stack) String() string {
+	return strings.Join(s, ".")
+}
+
+// MapFlattener defines the data structures and the implemetation of how to flatten a nested map.
+type MapFlattener struct {
```

Comment:
Hi @aslakknutsen ,
Did this to make it thread-safe based on a discussion with @kwk 



> ATTENTION: thread-safety
> 
> Having nestedKeyStack defined as a global variable to which other functions append or read from doesn't make this thread-safe. I suggest to create a struct and with a nestedKeyStack member and make the traverseMapAsDFS and Flatten functions methods of this struct.
>